fix(Tree): Match design spec to show/hide selectable indicator

assuncaocharles opened the pull request:
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

## BREAKING CHANGES

In master when we introduced the `selectable` feat for `Tree`, we were only showing indicator if it's expanded. #12362
Lately, as a fix for #14720 we merged #14916 

None of this behaviors match the design spec that says:

* Indicator is shown on hover and hidden on blur
* Selectable parent always show indicator when it's expanded. 

This PR introduces the behavior described above

![ANcx1OuK15](https://user-images.githubusercontent.com/8545105/93665393-5fe2b780-fa76-11ea-9b41-c3a183bf79e3.gif)

This PR also makes all items ( parent and child ) selectable by default in a selectable `Tree`. 


#### Focus areas to test

Indicator should alway appear on Mouse Hover and Focus and hide on Mouse Leave and Blur

layershifter commented on packages/fluentui/react-northstar/src/themes/teams/components/Tree/treeTitleStyles.ts:
```diff
@@ -1,34 +1,59 @@
+        }),
+        ...borderFocusStyles[':focus'],
+      },
+      ':focus-visible': {
+        ...borderFocusStyles[':focus-visible'],
+      },
```
```suggestion
      ':focus-visible': borderFocusStyles[':focus-visible'],
```

assuncaocharles commented:
@pompomon @layershifter Checked all screen changes, seems correct for me.

assuncaocharles commented:
@layershifter @pompomon @miroslavstastny please, check this one

layershifter commented:
> In master when we introduced the `selectable` feat for `Tree`,

Was this released in `0.51.0`? If no, it's not a breaking change for customers.

layershifter commented on packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:
```diff
@@ -131,15 +133,18 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
+
   const hasSubtreeItem = hasSubtree(props);
 
+  // const [showIndicator, setShowIndicator] = React.useState(false);
```
Cruft?

assuncaocharles replied:
Yes, removed

assuncaocharles commented on packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:
```diff
@@ -131,15 +133,18 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
+
   const hasSubtreeItem = hasSubtree(props);
 
+  // const [showIndicator, setShowIndicator] = React.useState(false);
```
```suggestion
```

assuncaocharles replied:
```suggestion
```

assuncaocharles commented on packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:
```diff
@@ -131,15 +133,17 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
+
   const hasSubtreeItem = hasSubtree(props);
 
+
```
```suggestion
```

assuncaocharles commented on packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:
```diff
@@ -131,13 +133,14 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
   } = props;
 
+  const selectableParent = hasSubtree && selectable;
+
```
```suggestion
```

assuncaocharles commented:
> > In master when we introduced the `selectable` feat for `Tree`,
> 
> Was this released in `0.51.0`? If no, it's not a breaking change for customers.

@layershifter We are breaking what we introduced in 0.49 in this PR #12362
